src/containerloops.jl

```diff
@@ -0,0 +1,240 @@
+import Base.keys
+import Base.values
+
+## These are the containers that can be looped over
+## The suffix 0 is for SortedDict and SortedMultiDict
+## The suffix 1 is for SortedSet.  No suffix means any of the
+## three.  
+
+typealias SCContainers0 Union(SortedDict, SortedMultiDict)
+typealias SCContainers Union(SCContainers0, SortedSet)
```

Comment:
Okay. Rather than suffixing types and functions with `0` or `1`, would you mind being slightly more descriptive?  It's great that you described the naming convention in a comment (and your descriptive comments overall are very much appreciated), but it would be nice to be able to look at the code and immediately understand what it means.

Of course, the names themselves do get pretty long at some point... perhaps an `SDD` prefix for `SortedDict`+`SortedMultiDict` and and `SS` for `SortedSet`?  I'm not wedded to this, so if you have a better idea, go for it.